sdk/communication/Azure.Communication.Common/README.md

```diff
@@ -69,10 +69,14 @@ that a valid token is acquired prior to executing the request.
 Optionally, you can enable proactive token refreshing where a fresh token will be acquired as soon as the
 previous token approaches expiry. Using this method, your requests are less likely to be blocked to acquire a fresh token:
 
+Optionally, you can adjust the time span before token expiry that will trigger the proactive refreshing of the token. For example, if the proactive refreshing is enabled ('refreshProactively' is true), setting the time span to 5 minutes means that 5 minutes before the cached token expires, the proactive refresh will request a new token by calling the 'tokenRefresher' callback.
+The default value is 10 minutes.
+
 ```C# Snippet:CommunicationTokenCredential_CreateRefreshableWithoutInitialToken
 using var tokenCredential = new CommunicationTokenCredential(
     new CommunicationTokenRefreshOptions(
         refreshProactively: true, // Indicates if the token should be proactively refreshed in the background or only on-demand
+        refreshTimeBeforeTokenExpiry: new TimeSpan(0, 0, 5, 0), // Proactive refreshing will be triggered 5 minutes before the token expires
```

Comment:
Have we considered making `refreshTimeBeforeTokenExpiry` a public property instead of a constructor argument similar to `AsyncTokenRefresher` and `InitialToken`? It's similar optional. I think it should be a property because usually a C# options bag only has the default constructor and properties which are all considered optional. But here we had to introduce a constructor because `tokenRefresher` is required.

I suggest to remove the new constructor and make the property public with a setter. Then the Contoso code would look like

```c#
using var tokenCredential = new CommunicationTokenCredential(
    new CommunicationTokenRefreshOptions(
        refreshProactively: true, // Indicates if the token should be proactively refreshed in the background or only on-demand
        tokenRefresher: cancellationToken => FetchTokenForUserFromMyServer("bob@contoso.com", cancellationToken))
    {
        RefreshTimeBeforeTokenExpiry = new TimeSpan(0, 0, 5, 0), // Proactive refreshing will be triggered 5 minutes before the token expires
        AsyncTokenRefresher = cancellationToken => FetchTokenForUserFromMyServerAsync("bob@contoso.com", cancellationToken)
    });
```

Sorry for not seeing this earlier.

Reply:
Here's the applicable guideline: https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-parameters

> The Options class is designed similarly to .NET custom attributes, where required service method parameters are modeled as Options class constructor parameters and get-only properties, and optional parameters are get-set properties.

Reply:
@DominikMe thank you so much for your comments and a huge impact on the improvements😊 Your comment is fixed for .net, should we change it as well for other languages?

Reply:
@DominikMe makes sense, thanks :)
@AikoBB if it's applicable to Java (check the guidelines) then yes. JS/TS works differently and Python's not supposed to have option bags.